From 076e85b79590aa8afbfe730a5120bf1332b17d2a Mon Sep 17 00:00:00 2001
From: Jonas Karlman <jonas@kwiboo.se>
Date: Sun, 13 Jul 2025 23:33:52 +0000
Subject: [PATCH 27/51] rockchip: odroid-go2: Use appropriate bootph props

GPIO devices are needed in U-Boot proper phase, sdmmc and sfc devices
are needed in SPL and pre-reloc phase.

Update bootph- props to match what boot phase devices are needed at.

Signed-off-by: Jonas Karlman <jonas@kwiboo.se>
---
 arch/arm/dts/rk3326-odroid-go2-u-boot.dtsi | 16 +++++++---------
 1 file changed, 7 insertions(+), 9 deletions(-)

diff --git a/arch/arm/dts/rk3326-odroid-go2-u-boot.dtsi b/arch/arm/dts/rk3326-odroid-go2-u-boot.dtsi
index df21747fad7..a27994e450c 100644
--- a/arch/arm/dts/rk3326-odroid-go2-u-boot.dtsi
+++ b/arch/arm/dts/rk3326-odroid-go2-u-boot.dtsi
@@ -27,22 +27,18 @@
 };
 
 &gpio0 {
-	bootph-all;
 	gpio-ranges = <&pinctrl 0 0 32>;
 };
 
 &gpio1 {
-	bootph-all;
 	gpio-ranges = <&pinctrl 0 32 32>;
 };
 
 &gpio2 {
-	bootph-all;
 	gpio-ranges = <&pinctrl 0 64 32>;
 };
 
 &gpio3 {
-	bootph-all;
 	gpio-ranges = <&pinctrl 0 96 32>;
 };
 
@@ -78,28 +74,30 @@
 };
 
 &sdmmc {
-	bootph-all;
+	bootph-pre-ram;
+	bootph-some-ram;
 
 	/* mmc to sram can't do dma, prevent aborts transferring TF-A parts */
 	u-boot,spl-fifo-mode;
 };
 
 &sfc {
-	bootph-all;
+	bootph-some-ram;
 };
 
 &{/spi@ff3a0000/flash@0} {
-	bootph-all;
+	bootph-pre-ram;
+	bootph-some-ram;
 };
 
 &uart1 {
-	clock-frequency = <24000000>;
 	bootph-all;
+	clock-frequency = <24000000>;
 };
 
 &uart2 {
-	clock-frequency = <24000000>;
 	bootph-all;
+	clock-frequency = <24000000>;
 };
 
 &xin24m {
-- 
2.34.1

